nats/bench_command.go

```diff
@@ -181,29 +270,69 @@ func (c *benchCmd) runPublisher(bm *bench.Benchmark, nc *nats.Conn, startwg *syn
 	donewg.Done()
 }
 
-func (c *benchCmd) runSubscriber(bm *bench.Benchmark, nc *nats.Conn, startwg *sync.WaitGroup, donewg *sync.WaitGroup) {
+func (c *benchCmd) runSubscriber(bm *bench.Benchmark, nc *nats.Conn, startwg *sync.WaitGroup, donewg *sync.WaitGroup, numMsg int) {
 	received := 0
 	ch := make(chan time.Time, 2)
 
-	sub, _ := nc.Subscribe(c.subject, func(msg *nats.Msg) {
+	mh := func(msg *nats.Msg) {
 		received++
+		if c.js && c.ack {
+			msg.Ack()
+		}
 		if received == 1 {
 			ch <- time.Now()
 		}
-		if received >= c.numMsg {
+		if received >= numMsg {
 			ch <- time.Now()
 		}
-	})
+	}
+	var sub *nats.Subscription
+
+	log.Printf("Starting subscriber, expecting %s messages", humanize.Comma(int64(numMsg)))
+
+	if c.js {
+		js, err := nc.JetStream()
+		if err != nil {
+			log.Fatalf("couldn't create jetstream context: %v", err)
+		}
+
+		if c.pull {
+			//sub, _ = js.SubscribeSync(c.subject)
+			sub, err = js.PullSubscribe(c.subject, JS_PULLCONSUMER_NAME)
+			if err != nil {
+				println("error pullsubscribe=" + err.Error())
+			}
+		} else {
+			sub, _ = nc.Subscribe(c.subject+".pushconsumer", mh)
+		}
+	} else {
+		sub, _ = nc.Subscribe(c.subject, mh)
+	}
 
 	sub.SetPendingLimits(-1, -1)
 	nc.Flush()
 	startwg.Done()
 
+	if c.js && c.pull {
+		for i := 0; i < numMsg; i++ {
+			msg, err := sub.NextMsg(time.Second * 3)
```

Comment:
For a pull consumer here have to call `Fetch(n)` where n is the batch number, could be 1 for a single message.